lib/ansible/modules/cloud/azure/GUIDELINES.md

```diff
@@ -0,0 +1,155 @@
+
+# Rules for Azure Ansible facts modules
+
+## Input Parameters
+
+Every facts module should at least contain following options. In most cases these options should be sufficient:
+
+|Option|Description|Optional|
+|------|-----------|--------|
+|resource_group|Resource group name|YES|
+|name|Resource Name|YES|
+|tags|Tags that should be matched|YES|
+
+Facts module should allow users to:
+- query all the resources in a given subscription (if possible)
+- query all the resources in a specific resource group (if possible)
+- query specific resource instance
+- filter resources by tags (if possible)
+
+## Return value rules
+
+Return values should be a composite of:
+- input of the main module
+    - structure should be preserved as closely as possible
+    - fields should be omited only if impossible or difficult to retrieve
+- additional read only fields assigned by the service, for instance:
+    - resource **id** field
+    - assigned **ip_address**
+    - any other fields that may be important to the user, like URLs etc.
+
+Following rules should be followed:
+- return structure should have as short name as possible, for instance **servers**
+- return structure should always be a **list of dictionaries**, even when single resource is queried
+- every dictionary should be as flat as possible, no deep structures
+- field names should be in **python_case** rather than camelCase
+- field values should be exactly the same as equivalent input values of the main module (preferably both **python_case**)
+- return structure should always contain resource **name**
+- return structure should always contain **resource_group** name
+
+Return values should be structured as in following example:
```

Comment:
could you add a real facts module link, like databases or webapp, so people could  refer to right sample,  and have full code